agent/src/com/cloud/agent/Agent.java

```diff
@@ -697,16 +697,34 @@ private Answer setupDirectDownloadCertificate(SetupDirectDownloadCertificate cmd
 
         final String keyStoreFile = agentFile.getParent() + "/" + KeyStoreUtils.KS_FILENAME;
 
-        String cerFile = agentFile.getParent() + "/" + certificateName + ".cer";
-        Script.runSimpleBashScript(String.format("echo '%s' > %s", certificate, cerFile));
+        String cerFile = agentFile.getParent() + "/" + "CSCERTIFICATE-" + certificateName;
```

Comment:
what is the file prefix for?

Reply:
This is to make sure the certificate is created into a non existing file